elasticsearch/includes/classes/class-health-job.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace Automattic\VIP\Elasticsearch;
+
+use Automattic\VIP\Elasticsearch\Health as Health;
+
+class HealthJob {
+
+	/**
+	 * The name of the scheduled cron event to run the health check
+	 */
+	const CRON_EVENT_NAME = 'vip_search_healthcheck';
+
+	/**
+	 * Custom cron interval name
+	 */
+	const CRON_INTERVAL_NAME = 'vip_search_healthcheck_interval';
+
+	/**
+	 * Custom cron interval value
+	 *
+	 * 30 minutes in seconds
+	 */
+	const CRON_INTERVAL = 60 * 30;
+
+	/**
+	 * Initialize the job class
+	 *
+	 * @access	public
+	 */
+	public function init() {
+		// Add the custom cron schedule
+		add_filter( 'cron_schedules', [ $this, 'filter_cron_schedules' ], 10, 1 );
+
+		$this->schedule_job();
+	}
+
+	/**
+	 * Schedule health check job
+	 *
+	 * Add the event name to WP cron schedule and then add the action
+	 */
+	public function schedule_job() {
+		if ( ! wp_next_scheduled( self::CRON_EVENT_NAME )  ) {
+			wp_schedule_event( time(), self::CRON_INTERVAL_NAME, self::CRON_EVENT_NAME );
+		}
+
+		add_action( self::CRON_EVENT_NAME, [ $this, 'check_health' ] );
+	}
+
+	/**
+	 * Filter `cron_schedules` output
+	 *
+	 * Add the custom interval to WP cron schedule
+	 *
+	 * @param		array	$schedule
+	 *
+	 * @return	mixed
+	 */
+	public function filter_cron_schedules( $schedule ) {
+		if ( isset( $schedule[ self::CRON_INTERVAL_NAME ] ) ) {
+			return $schedule;
+		}
+
+		// Not actually five minutes; we want it to run faster though to get through everything.
+		$schedule[ self::CRON_INTERVAL_NAME ] = [
+			'interval' => self::CRON_INTERVAL,
+			'display' => __( 'VIP Search Healthcheck time interval' ),
+		];
+
+		return $schedule;
+	}
+
+	/**
+	 * Check ElasticSearch index health
+	 */
+	public function check_health() {
+		$user_results = Health::validate_index_users_count();
+
+
+		$this->process_results( $user_results );
+
+		$post_results = Health::validate_index_posts_count();
+
+		$this->process_results( $user_results );
+	}
+
+	/**
+	 * Process the health check result
+	 *
+	 * @access	protected
+	 * @param		array		$result		Array of results from Health index validation
+	 */
+	protected function process_results( $result ) {
+		// If there's an error, alert
+		if( array_key_exists( 'error', $results ) ) {
+			wpcom_vip_irc(
```

Comment:
Aside: we may want to consider a few things related to this monitoring (outside this PR):

1) do we want to monitor and alert for non-production sites?
2) do we want a way to turn off alerts globally?
3) How do we monitor that this is still active and running (some sort of heartbeat)?

Reply:
> do we want to monitor and alert for non-production sites?
I don't think we should. This could generate a lot noise that buries problems in production sites. 

> do we want a way to turn off alerts globally?
You mean a constant that can disable alerts for a site?

> How do we monitor that this is still active and running (some sort of heartbeat)?
We can do that but I am worried about the noise. Perhaps it goes to a different channel?